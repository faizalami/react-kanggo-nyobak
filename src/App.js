import { Routes, Route, Navigate } from 'react-router-dom';
import Layout from './components/Layout';
import Index from './pages/products/Index';
import Details from './pages/products/Detail';
import Form from './pages/products/Form';
import NotFound from './pages/NotFound';

function App () {
  return (
    <Routes>
      <Route path="/" element={<Navigate to="/product"/>}/>
      <Route path="product" element={<Layout title="Product"/>}>
        <Route index element={<Index/>}/>
        <Route path=":id" element={<Details/>}/>
        <Route path="add" element={<Form/>}/>
        <Route path="edit/:id" element={<Form/>}/>
      </Route>
      <Route
        path="*"
        element={<NotFound/>}
      />
    </Routes>
  );
}

export default App;
