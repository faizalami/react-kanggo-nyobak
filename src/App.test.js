import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

test('App matches snapshot', () => {
  expect(render(
    <MemoryRouter>
      <App/>
    </MemoryRouter>,
  )).toMatchSnapshot();
});
