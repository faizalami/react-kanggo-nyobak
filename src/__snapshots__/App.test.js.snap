// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App App matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  background-color: #1e293b;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 80rem;
  margin-left: 1rem;
  margin-right: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-2 {
  margin-top: 0rem;
  margin-bottom: 0rem;
  margin-right: 1rem;
  color: white;
  font-size: 1.5rem;
}

.emotion-3 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #111827;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #4f46e5;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #d1d5db;
}

.emotion-5 {
  background-color: white;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 80rem;
  margin-left: 1rem;
  margin-right: 1rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

@media screen and (min-width: 1024px) {
  .emotion-6 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-7 {
  margin-top: 0rem;
  margin-bottom: 0rem;
  margin-right: 1rem;
  color: #111827;
  font-size: 2rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 80rem;
  margin-left: 1rem;
  margin-right: 1rem;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

@media screen and (min-width: 1024px) {
  .emotion-8 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-10 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #4f46e5;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
}

.emotion-10:hover,
.emotion-10:focus {
  background-color: #4338ca;
}

.emotion-11 {
  width: 1rem;
  height: 1rem;
  margin-right: 0.25rem;
}

.emotion-12 {
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  gap: 1rem;
  border: 4px dashed #d1d5db;
  border-radius: 0.5rem;
  padding: 1rem;
  width: 100%;
  margin-top: 0.75rem;
}

@media screen and (min-width: 1024px) {
  .emotion-12 {
    grid-template-columns: repeat(4, minmax(0, 1fr));
  }
}

<body>
    <div>
      <nav
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <h1
            class="emotion-2"
          >
            MyWeb
          </h1>
          <a
            class="emotion-3"
            href="/product"
          >
            Product
          </a>
          <a
            class="emotion-4"
            href="https://faizalami.github.io"
            rel="noreferrer"
            target="_blank"
          >
            About Me
          </a>
        </div>
      </nav>
      <header
        class="emotion-5"
      >
        <div
          class="emotion-6"
        >
          <h2
            class="emotion-7"
          >
            Product
          </h2>
        </div>
      </header>
      <main
        class="emotion-8"
      >
        <div
          class="emotion-9"
        >
          <a
            class="emotion-10"
            href="/product/add"
          >
            <svg
              class="emotion-11"
            >
              plus.svg
            </svg>
             New Product
          </a>
          <article
            class="emotion-12"
            cols="1"
          />
        </div>
      </main>
      ,
    </div>
  </body>,
  "container": .emotion-0 {
  background-color: #1e293b;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 80rem;
  margin-left: 1rem;
  margin-right: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-2 {
  margin-top: 0rem;
  margin-bottom: 0rem;
  margin-right: 1rem;
  color: white;
  font-size: 1.5rem;
}

.emotion-3 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #111827;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #4f46e5;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #d1d5db;
}

.emotion-5 {
  background-color: white;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 80rem;
  margin-left: 1rem;
  margin-right: 1rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

@media screen and (min-width: 1024px) {
  .emotion-6 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-7 {
  margin-top: 0rem;
  margin-bottom: 0rem;
  margin-right: 1rem;
  color: #111827;
  font-size: 2rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 80rem;
  margin-left: 1rem;
  margin-right: 1rem;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

@media screen and (min-width: 1024px) {
  .emotion-8 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-10 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: white;
  background-color: #4f46e5;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
}

.emotion-10:hover,
.emotion-10:focus {
  background-color: #4338ca;
}

.emotion-11 {
  width: 1rem;
  height: 1rem;
  margin-right: 0.25rem;
}

.emotion-12 {
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  gap: 1rem;
  border: 4px dashed #d1d5db;
  border-radius: 0.5rem;
  padding: 1rem;
  width: 100%;
  margin-top: 0.75rem;
}

@media screen and (min-width: 1024px) {
  .emotion-12 {
    grid-template-columns: repeat(4, minmax(0, 1fr));
  }
}

<div>
    <nav
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <h1
          class="emotion-2"
        >
          MyWeb
        </h1>
        <a
          class="emotion-3"
          href="/product"
        >
          Product
        </a>
        <a
          class="emotion-4"
          href="https://faizalami.github.io"
          rel="noreferrer"
          target="_blank"
        >
          About Me
        </a>
      </div>
    </nav>
    <header
      class="emotion-5"
    >
      <div
        class="emotion-6"
      >
        <h2
          class="emotion-7"
        >
          Product
        </h2>
      </div>
    </header>
    <main
      class="emotion-8"
    >
      <div
        class="emotion-9"
      >
        <a
          class="emotion-10"
          href="/product/add"
        >
          <svg
            class="emotion-11"
          >
            plus.svg
          </svg>
           New Product
        </a>
        <article
          class="emotion-12"
          cols="1"
        />
      </div>
    </main>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
